hive-metastore/src/main/java/org/apache/iceberg/hive/HiveViewOperations.java

```diff
@@ -0,0 +1,287 @@
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.exceptions.CommitStateUnknownException;
+import org.apache.iceberg.exceptions.NoSuchIcebergViewException;
+import org.apache.iceberg.exceptions.NoSuchViewException;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.util.MetastoreOperationsUtil;
+import org.apache.iceberg.view.BaseViewOperations;
+import org.apache.iceberg.view.ViewMetadata;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Hive implementation of Iceberg ViewOperations. */
+final class HiveViewOperations extends BaseViewOperations implements HiveOperationsBase {
+  private static final Logger LOG = LoggerFactory.getLogger(HiveViewOperations.class);
+
+  private final String fullName;
+  private final String database;
+  private final String viewName;
+  private final FileIO fileIO;
+  private final ClientPool<IMetaStoreClient, TException> metaClients;
+  private final long maxHiveTablePropertySize;
+
+  HiveViewOperations(
+      Configuration conf,
+      ClientPool<IMetaStoreClient, TException> metaClients,
+      FileIO fileIO,
+      String catalogName,
+      TableIdentifier viewIdentifier) {
+    String dbName = viewIdentifier.namespace().level(0);
+    this.metaClients = metaClients;
+    this.fileIO = fileIO;
+    this.fullName = BaseMetastoreCatalog.fullTableName(catalogName, viewIdentifier);
+    this.database = dbName;
+    this.viewName = viewIdentifier.name();
+    this.maxHiveTablePropertySize =
+        conf.getLong(HIVE_TABLE_PROPERTY_MAX_SIZE, HIVE_TABLE_PROPERTY_MAX_SIZE_DEFAULT);
+  }
+
+  @Override
+  public void doRefresh() {
+    String metadataLocation = null;
+    Table table = null;
+
+    try {
+      table = metaClients.run(client -> client.getTable(database, viewName));
+      HiveOperationsBase.validateTableOrViewIsIceberg(table, fullName);
+      metadataLocation =
+          table.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP);
+
+    } catch (NoSuchObjectException e) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchViewException("View does not exist: %s.%s", database, viewName);
+      }
+    } catch (TException e) {
+      String errMsg =
+          String.format("Failed to get view info from metastore %s.%s", database, viewName);
+      throw new RuntimeException(errMsg, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
+    }
+
+    if (table != null && !table.getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())) {
+      disableRefresh();
+    } else {
+      refreshFromMetadataLocation(metadataLocation);
+    }
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  @Override
+  public void doCommit(ViewMetadata base, ViewMetadata metadata) {
```

Comment:
What are the main differences between this method and the `HiveTableOperations.doCommit`? How much of the code is the same?

Reply:
The main reason is for Hive Table we are using commit lock. For View we are not using any lock mechanism.

Reply:
Why not use commit lock for view creation, modification?
We could have the same concurrency issues here, that with the tables.

Reply:
I thought adding lock for view creation is expensive and unnecessary. Since view is more like a read definition. With View we are not doing any data write. 
Please share your thoughts.

Reply:
Iceberg tables and views should have a linear history by design. If it is not enforced somehow (for example by locks), then it could cause unexpected results